src/App.tsx

```diff
@@ -116,75 +126,97 @@ const hexRegex = /^#[a-f0-9]{6}$/gi;
 
 // TODO: add an error boundary for parse fails
 const Editor: Component<{ currentFile: FileSystemFileHandle }> = (props) => {
-  const [contents, { mutate }] = createResource(async () => {
+  const [obj, setObject] = createStore<Record<string, unknown>>({});
+  let lastUpdate = Date.now();
+  createEffect(async () => {
     const file = await props.currentFile.getFile();
     const text = await file.text();
-    console.log(parseJsonFile(text));
-    return text;
+    const json = JSON.parse(text);
+    setObject(reconcile(json));
+  });
+
+  createEffect(() => {
```

Comment:
I think we can get away with not using an effect here? see https://www.solidjs.com/tutorial/lifecycles_oncleanup?solved

I might be wrong, though. this might be necessary to detect changes in the props?

same with the effect above.

Reply:
interesting point. i thinkk we need both of these `createEffect`s though since we need these effects to rerun every time `props.currentFile` changes. that `onCleanup` example is simpler because the `setInterval` doesn't depend on props or state.